src/frontend/src/pages/FinancePage/RefundsSection.tsx

```diff
@@ -0,0 +1,153 @@
+  },
+  [`& .${linearProgressClasses.bar}`]: {
+    borderRadius: 5,
+    backgroundColor: '#ef4345'
+  }
+}));
+
+const createRefundData = (date: Date, amount: number, recipient: User) => {
+  return { date, amount, recipient };
+};
+
+interface RefundTableProps {
+  currentUserRequests: ReimbursementRequest[];
+  allRequests?: ReimbursementRequest[];
+}
+
+const Refunds = ({ currentUserRequests, allRequests }: RefundTableProps) => {
+  const [value, setValue] = useState(0);
+  const user = useCurrentUser();
+
+  const { data, isLoading, isError, error } = useCurrentUserReimbursements();
+  const { data: allData, isLoading: allIsLoading, isError: allIsError, error: allError } = useAllReimbursements();
+  const theme = useTheme();
+
+  if (user.isFinance && allIsError) return <ErrorPage message={allError?.message} />;
+  if (isError) return <ErrorPage message={error?.message} />;
+  if ((user.isFinance && (allIsLoading || !allData)) || isLoading || !data) return <LoadingIndicator />;
+
+  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
+    setValue(newValue);
+  };
+
+  const rows = (allData && value === 1 ? allData : data).map((row: Reimbursement) =>
+    createRefundData(row.dateCreated, row.amount, row.userSubmitted)
+  );
+  const totalReceived = (allData && value === 1 ? allData : data).reduce(
+    (accumulator: number, currentVal: Reimbursement) => accumulator + currentVal.amount,
+    0
+  );
+  const totalOwed = (allRequests && value === 1 ? allRequests : currentUserRequests).reduce(
+    (accumulator: number, currentVal: ReimbursementRequest) => accumulator + currentVal.totalCost,
+    0
+  );
+  const percentRefunded = (totalReceived / totalOwed) * 100;
+
+  return (
+    <Box sx={{ bgcolor: theme.palette.background.paper, width: '100%', borderRadius: '8px 8px 8px 8px', boxShadow: 1 }}>
+      <AppBar sx={{ borderRadius: '8px 8px 0 0' }} position="static">
+        <Tabs value={value} onChange={handleChange} indicatorColor="secondary" textColor="inherit" variant="fullWidth">
+          <Tab
+            sx={{ borderRadius: '8px 8px 0 0', fontWeight: 700, pointerEvents: user.isFinance ? 'auto' : 'none' }}
+            label="My Refunds"
+            value={0}
+          />
+          {user.isFinance && (
+            <Tab sx={{ borderRadius: '8px 8px 0 0', fontWeight: 700 }} label="All Club Refunds" value={1} />
+          )}
+        </Tabs>
+      </AppBar>
+      <Box
+        sx={{
+          backgroundColor: theme.palette.background.paper,
+          width: '100%',
+          padding: '10px',
+          borderRadius: '0 0 8px 8px'
+        }}
+      >
+        <Box sx={{ padding: '30px' }}>
+          <Box sx={{ display: 'flex', flexDirection: 'horizontal', justifyContent: 'space-between', paddingX: '30px' }}>
+            <Stack sx={{ alignItems: 'center' }}>
+              <Typography textAlign="center" sx={{ fontWeight: 700 }}>
```

Comment:
make this a component you use it multiple times

Reply:
sure